### PR TITLE
Add colors decoded from a hexadecimal string

### PR DESCRIPTION
**Pull request by Joannis**

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
Thank you @Joannis, I've just merged a very similar PR #69 this morning, which supports integer literals, which can be hex too:

```swift
let color: Color = 0xFF00FF
```

Do you think there's a benefit in supporting a similar syntax for string hex values? In my opinion the downside is that the string-based initializer becomes optional and requires users to unwrap that value before passing it as an argument to props, while with an integer literal we're always sure it's a valid unsigned 32-bit value. But maybe I'm missing something 🤔

**Comment by MaxDesiatov** [comment]
BTW, CI build fails as you need to run `swiftformat .` in your root clone directory. I'm thinking we should use something like [Danger](https://danger.systems/swift/) to report these kind of issues, PRs to enable that are very welcome. I've previously asked houndci.com support if they can integrate with SwiftFormat somehow, but got no response so far.

**Comment by Joannis** [comment]
I noticed, and I think your solution is more common in addition to it being performant. The reason I wanted to add an initializer like this is for color configs and colors loaded from APIs. It's a bit more niche for sure, so I'd completely understand it if it shouldn't be a part of the main library.

On the 32-bit integer notice. You're throwing away the highest 8 bits which could hold an alpha value. I'd `assert()` that those **aren't** set, so that there's no room for confusion or unclarity.

**Comment by MaxDesiatov** [comment]
@Joannis This makes sense, loading from configs is a great use case. 

Can you please run `swiftformat .` with the latest SwiftFormat version in the root of your git clone and fix remaining linter violations highlighted by Hound?

Have you ever seen alpha encoded in hex? I wonder if it's better to decode it when it's available rather than assert. OTOH, default alpha value is 1, while absent bits mean 0, so there is no good way to distinguish between 0 alpha and implied alpha 1, if I understand this correctly?

Thanks!

File: Sources/Tokamak/Components/Props/Color.swift [hunk 1]
```diff
@@ -46,3 +46,71 @@ extension Color: ExpressibleByIntegerLiteral {
...
+    alpha = 1
+    space = .sRGB
+  }
+}
+
+private let lowerRadix16table: [UInt8] = [
+  0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46,
+]
+private let upperRadix16table: [UInt8] = [
+  0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
```
[the remark this answers is not in the document]
Comment (Joannis) [reply Sources/Tokamak/Components/Props/Color.swift]: @MaxDesiatov what do you think would be the cleanest way in this scenario?

[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Sources/Tokamak/Components/Props/Color.swift]: I guess there's no specific meaning in items being on one line, could it be then `0x3*` on the first line and `0x6*` on the second?

[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Sources/Tokamak/Components/Props/Color.swift]: and same for `lowerRadix16table` and its `0x3*` and `0x4*` items?

[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Sources/Tokamak/Components/Props/Color.swift]: I'd also appreciate if you put a link to the bit shuffling algorithm you're using it if it's described anywhere or even more future-proof option would be describing the gist of it in the comments. Thank you!

**Comment by Joannis** [comment]
Hound will be the bane of my existence...

**Comment by Joannis** [comment]
I don't think I agree with trimming down comments so that they fit on 80 characters. It feels like I'm fighting against the natural length of a human sentence. Especially considering there's code indentation there.

**Comment by MaxDesiatov** [comment]
@Joannis no probs, up to you. I personally prefer to stick to 80 for everything as it improves accessibility for people who use large fonts due to vision problems or people who don't have external wide monitors or read diffs on phones/tablets etc.

**Comment by MaxDesiatov** [comment]
And yes, it's more of a problem because of IDEs and GitHub diff viewer, which don't autowrap long lines cleanly

**Comment by Joannis** [comment]
I've never taken that use case as a very strict/serious one 🙈
But yeah, I tend to think of a maximum line length somewhere along the lines of 120 characters.

**Comment by MaxDesiatov** [comment]
BTW, do you know if SwiftFormat supports separate settings for comments and code lines? I think I wouldn't mind relaxing comment line length if you find wrapping comment lines not working for you

**Comment by Joannis** [comment]
You can, using `- ignores_comments`. Shall I add that? It restricts comments to 120 character lines.

**Pull request merged**